src/control/cmd/daos/property_test.go

```diff
@@ -0,0 +1,20 @@
+//
+// (C) Copyright 2022 Intel Corporation.
+//
+// SPDX-License-Identifier: BSD-2-Clause-Patent
+//
+
+package main
+
+import "testing"
+
+// The actual test functions are in property_ctest.go file so that they can use cgo (import "C").
```

Comment:
I've seen this technique with the cgo tests in the past and it's a bit of a hack because the test functions are exposed as part of the package as opposed to being hidden from the package symbols when defined in a `*_test.go` file. It is a workaround for the known limitation in CGO. We also do similar hacks by exposing test mocks in `mocks.go` files (and other regular package files which is bad) so you are not setting a precedent and therefore I think it's okay.

Reply:
For the record, I don't think this would be considered a hack, and it's different than our use of mocks (which are exported because there are some tests which cross package boundaries). The test function bodies defined in `property_ctest.go` are unexported (start with a lowercase letter), and these test functions are in a `_test.go` file so also not added to the exported package symbols. This solution for getting around cgo testing limitations is perfectly fine, IMO.